examples/reading_events.rs

```diff
@@ -0,0 +1,188 @@
+    ReadStreamOptions,
+};
+use futures::TryStreamExt;
+use serde::{Deserialize, Serialize};
+use std::error::Error;
+use uuid::Uuid;
+
+#[derive(Serialize, Deserialize, Debug)]
+struct TestEvent {
+    pub id: String,
+    pub important_data: String,
+}
+
+type Result<A> = std::result::Result<A, Box<dyn Error>>;
+
+pub async fn read_from_stream(client: &Client) -> Result<()> {
+    // region read-from-stream
+    let result = client
+        .read_stream("some-stream", &Default::default(), All)
+        .await?;
+    // endregion read-from-stream
+
+    // region iterate-stream
+    if let Some(mut events) = result.ok() {
+        while let Some(event) = events.try_next().await? {
+            let test_event = event.get_original_event().as_json::<TestEvent>()?;
+
+            println!("Event> {:?}", test_event);
+        }
+    }
+    // endregion iterate-stream
+
+    Ok(())
+}
+
+pub async fn read_from_stream_position(client: &Client) -> Result<()> {
+    // region read-from-position
+    let options = ReadStreamOptions::default().origin(Origin::Point(10));
+    let result = client.read_stream("some-stream", &options, 20).await?;
+    // endregion read-from-position
+
+    // region iterate-stream
+    if let Some(mut events) = result.ok() {
+        while let Some(event) = events.try_next().await? {
+            let test_event = event.get_original_event().as_json::<TestEvent>()?;
+
+            println!("Event> {:?}", test_event);
+        }
+    }
+    // endregion iterate-stream
+
+    Ok(())
+}
+
+pub async fn read_stream_overriding_user_credentials(client: &Client) -> Result<()> {
+    // region overriding-user-credentials
+    let options = ReadStreamOptions::default()
+        .origin(Origin::Start)
+        .authenticated(Credentials::new("admin", "changeit"));
+
+    let result = client.read_stream("some-stream", &options, All).await?;
+    // endregion overriding-user-credentials
+    Ok(())
+}
+
+pub async fn read_from_stream_position_check(client: &Client) -> Result<()> {
+    // region checking-for-stream-presence
+    let options = ReadStreamOptions::default().origin(Origin::Point(10));
+
+    let result = client.read_stream("some-stream", &options, All).await?;
+
+    match result {
+        ReadResult::Ok(mut events) => {
+            while let Some(event) = events.try_next().await? {
+                let test_event = event.get_original_event().as_json::<TestEvent>()?;
+
+                println!("Event> {:?}", test_event);
+            }
+        }
+
+        ReadResult::StreamNotFound(stream_name) => {
+            println!("Stream not found: {}", stream_name);
+
+            return Ok(());
+        }
+    }
+    // endregion checking-for-stream-presence
+    Ok(())
+}
+
+pub async fn read_stream_backwards(client: &Client) -> Result<()> {
+    // region reading-backwards
+    let options = ReadStreamOptions::default().origin(Origin::End);
```

Comment:
**FIX:** In other clients we're using the `Direction` name. Could you rename `Origin` to `Direction`?

Reply:
**CONCERN:** Setting direction to backwards seems to be missing here.

Reply:
We could also add dedicated handlers as in Java, so e.g. `from_end`, `from_start`.

Reply:
`from_end` and `from_start` doesn't comply to Rust naming convention hence why I changed it afterwards (when naming `from_*`, the method must not use `self`). This why I switch to this origin naming.

I find `StreamPosition` confusing because with have `Position` for `$all` stream while a regular stream uses `u64`.

Reply:
I agree on showing the direction even if in this situation it sets to `backwards` by default.

Reply:
Done